packages/cli/src/lib/implementation/core-config.options.ts

```diff
@@ -26,14 +31,17 @@ export function yargsPersistConfigOptionsDefinition(): Record<
     'persist.outputDir': {
       describe: 'Directory for the produced reports',
       type: 'string',
+      default: PERSIST_OUTPUT_DIR,
     },
     'persist.filename': {
       describe: 'Filename for the produced reports.',
       type: 'string',
+      default: PERSIST_FILENAME,
     },
     'persist.format': {
       describe: 'Format of the report output. e.g. `md`, `json`',
       type: 'array',
+      default: PERSIST_FORMAT,
     },
```

Comment:
I'm afraid that adding defaults to the CLI args introduces a bug - **persist options can't be customized via config file**. The precedence is supposed to be `explicit CLI arg > explicit config prop > default (if applicable)`, but setting the defaults at Yargs level means they're handled as explicit CLI args. Config middleware does `persist: {  ...importedRc.persist, ...cliOptions.persist }`, so treats explicit and implicit CLI options the exact same, and because everything under `persist` has defaults, values from config file are always overridden.

Here is an example of where the bug occurs. You'll notice that although I didn't specify `persist.outputDir` or `persist.format` in the CLI, my custom values in `code-pushup.config.ts` are overridden by defaults.
![image](https://github.com/code-pushup/cli/assets/34691111/4f6401b0-823c-4c2b-be9d-385b8b0dd749)

And you'll notice it works correctly in the main branch, so the bug is introduced in this PR.
![image](https://github.com/code-pushup/cli/assets/34691111/f07c0dd0-bf04-458c-9279-04e1e13df817)

Reply:
Dang! So then lets circle back to the config option of `yargs`. I would really like to use a more maintainable way to handle CIL settings.

Reply:
Update:
I refactored it back and added tests. 
Will test other options in the future.